cocos2d/core/renderer/RendererWebGL.js

```diff
@@ -47,8 +47,34 @@ cc.rendererWebGL = {
             i,
             len;
         var context = ctx || cc._renderContext;
-        for (i = 0, len = locCmds.length; i < len; i++) {
-            locCmds[i].rendering(context);
+        
+        //prepare batching
+        for (i = 0, len = locCmds.length; i< len; ++i) 
+        {
+            var cmd = locCmds[i];
+            if(!cmd._batched && cmd.configureBatch) //may be set to true by processed cmds during this loop 
+            {
+                cmd.configureBatch(locCmds,i);
```

Comment:
I didn't see the implementation of this function, I assume it should be implemented in cc.Sprite.WebGLRenderCmd. Normally it should set commands' _batching and _batched property by comparing their type and rendering data.